lms/djangoapps/class_dashboard/dashboard_data.py

```diff
@@ -348,9 +383,21 @@ def get_d3_section_grade_distrib(course_id, section):
     for problem in problem_set:
         stack_data = []
 
-        if problem in grade_distrib:  # Some problems have no data because students have not tried them yet.
-            max_grade = float(grade_distrib[problem]['max_grade'])
-            for (grade, count_grade) in grade_distrib[problem]['grade_distrib']:
+        grade_distribution = []
+
+        try:
+            module = client.modules(course_id, problem)
+            grade_distribution = module.grade_distribution()
+        except NotFoundError:
+            pass
+
+        # Only problems with recorded scores have had student submissions.
+        if len(grade_distribution) > 0:
```

Comment:
Since we're initializing `grade_distribution` above, couldn't we remove the `if len(...)` bit, unindent the block, and then just jump straight into the `for score in grade_distribution:` bit? If the list is empty, it should just pass on through.